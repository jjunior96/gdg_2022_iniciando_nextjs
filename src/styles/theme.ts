export default {
  grid: {
    container: '96rem',
    gutter: '3.2rem'
  },
  components: {
    sidebar: '20rem',
    header: '8rem'
  },
  border: {
    radius: '0.4rem'
  },
  box: {
    shadow: '0 8px 8px rgba(0, 0, 0, 0.2)'
  },
  font: {
    family:
      "Montserrat, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xxxxsmall: '0.8rem',
      xxxsmall: '1.2rem',
      xxsmall: '1.4rem',
      xsmall: '1.6rem',
      small: '2rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      xxxlarge: '5.6rem',
      huge: '6.4rem'
    }
  },
  colors: {
    primary: '#24A66B',
    secundary: '#5855E9',
    white: '#FAFAFA',
    gray_50: '#f0efeb',
    gray_100: '#D9D9D9',
    gray_200: '#91AAB4',
    gray_300: '#3E606F',
    gray_450: '#193441',
    gray_500: '#343a40',
    yellow_300: '#F1C40F',
    purple_100: '#7F57DF',
    purple_300: '#5855E9',
    green: '#24A66B',
    red: '#E63737',
    success: '#24A66B',
    error: '#E63737',
    info: '#189CEF',
    black: '#0D0D0D'
  },
  spacings: {
    xxxsmall: '0.8rem',
    xxsmall: '1rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  modal: {
    sizes: {
      small: '40rem',
      medium: '48rem',
      large: '54rem'
    }
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const;
