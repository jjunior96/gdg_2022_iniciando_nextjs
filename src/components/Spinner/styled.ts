import { VscLoading as LoadingIcon } from 'react-icons/vsc';

import styled, { css, DefaultTheme, keyframes } from 'styled-components';

import theme from 'styles/theme';

import { SpinnerProps } from '.';

const loadingAnimation = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const containerModifiers = {
  small: (theme: DefaultTheme) => css`
    width: ${theme.font.sizes.small};
  `,

  medium: (theme: DefaultTheme) => css`
    width: ${theme.font.sizes.xlarge};
  `,

  large: (theme: DefaultTheme) => css`
    width: ${theme.font.sizes.huge};
  `
};

export const Container = styled.div<SpinnerProps>`
  ${({ theme, size }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    margin: 0 auto;

    ${!!size && containerModifiers[size](theme)};

    z-index: ${theme.layers.alwaysOnTop};
  `}
`;

export const IconLoading = styled(LoadingIcon).attrs({
  color: theme.colors.green
})`
  width: 100%;

  animation: ${loadingAnimation} 1s linear infinite;
`;
