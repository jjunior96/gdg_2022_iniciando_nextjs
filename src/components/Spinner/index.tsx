import * as S from './styled';

export type SpinnerProps = {
  size?: 'small' | 'medium' | 'large';
};

const Spinner = ({ size = 'medium' }: SpinnerProps) => {
  return (
    <S.Container size={size} data-testid="spinner">
      <S.IconLoading data-testid="spinner-icon" />
    </S.Container>
  );
};

export default Spinner;
