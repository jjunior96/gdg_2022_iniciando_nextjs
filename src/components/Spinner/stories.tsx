import { Story, Meta } from '@storybook/react/types-6-0';

import Spinner, { SpinnerProps } from '.';

export default {
  title: 'Spinner',
  component: Spinner,
  decorators: [
    (Story) => (
      <div style={{ width: '40rem', margin: '0 auto', padding: '1rem 0 1rem' }}>
        <Story />
      </div>
    )
  ]
} as Meta;

export const Default: Story<SpinnerProps> = (args) => <Spinner {...args} />;
