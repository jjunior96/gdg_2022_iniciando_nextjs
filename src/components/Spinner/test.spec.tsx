import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/__config__/helpers';

import theme from 'styles/theme';

import Spinner from '.';

describe('<Spinner />', () => {
  it('should render the Spinner component', () => {
    const { container } = renderWithTheme(<Spinner data-testid="spinner" />);

    expect(screen.getByTestId('spinner')).toHaveStyle({
      width: theme.font.sizes.xlarge
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Spinner with size small', () => {
    renderWithTheme(<Spinner size="small" />);

    expect(screen.getByTestId('spinner')).toHaveStyle({
      width: theme.font.sizes.small
    });
  });

  it('should render the Spinner with size large', () => {
    renderWithTheme(<Spinner size="large" />);

    expect(screen.getByTestId('spinner')).toHaveStyle({
      width: theme.font.sizes.huge
    });
  });
});
