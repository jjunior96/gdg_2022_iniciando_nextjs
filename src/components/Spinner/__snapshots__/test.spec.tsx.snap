// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render the Spinner component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  width: 4rem;
  z-index: 50;
}

.c1 {
  width: 100%;
  -webkit-animation: eoUyJr 1s linear infinite;
  animation: eoUyJr 1s linear infinite;
}

<div
  class="c0"
  data-testid="spinner"
>
  <svg
    class="c1"
    color="#24A66B"
    data-testid="spinner-icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    style="color: rgb(36, 166, 107);"
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M13.917 7A6.002 6.002 0 0 0 2.083 7H1.071a7.002 7.002 0 0 1 13.858 0h-1.012z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
